test/test_workspace.py

```diff
@@ -0,0 +1,131 @@
+"""Test that workspace operations act like we expect them to."""
+import pytest
+
+from uuid import uuid4
+from multinet.db import (
+    create_workspace,
+    delete_workspace,
+    rename_workspace,
+    workspace_exists,
+    workspace_mapping,
+)
+
+
+@pytest.fixture
+def generated_workspace():
+    """Create a workspace, and yield the name of the workspace."""
+
+    workspace_name = uuid4().hex
+
+    create_workspace(workspace_name)
+    return workspace_name
+
+
+@pytest.fixture
+def handled_workspace(generated_workspace):
+    """
+    Create a workspace, and yield the name of the workspace.
+
+    On teardown, deletes the workspace.
+    """
+    yield generated_workspace
+
+    delete_workspace(generated_workspace)
+    workspace_mapping.cache_clear()
```

Comment:
The `workspace_mapping` cache is invalidated by the delete_workspace function so you might not need to clear the cache manually

Reply:
Good point